### PR TITLE
Fix sigabort on x86_64 HUP

### PR DESCRIPTION
**Pull request by alexgg**
See: https://github.com/balena-os/meta-balena/issues/2589

### CODE REVIEW

**Comment by alexgg** [comment]
Needs to be thoroughly tested. Also, the container build needs to be fixed.

**Comment by klutchell** [comment]
I will try some balena-intel builds with this branch and start testing.

File: go.mod [hunk 1]
```diff
@@ -2,26 +2,22 @@ module github.com/balena-os/hostapp
...
+	github.com/opencontainers/image-spec v1.0.2 // indirect
+	github.com/opencontainers/runc v1.1.0 // indirect
+	github.com/opencontainers/selinux v1.10.1 // indirect
+	github.com/vbatts/tar-split v0.11.2 // indirect
+	golang.org/x/net v0.0.0-20211216030914-fe4d6282115f // indirect
+	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad
+	golang.org/x/text v0.3.7 // indirect
+	google.golang.org/genproto v0.0.0-20211208223120-3a66f561d7aa // indirect
+	google.golang.org/grpc v1.45.0 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
```
Comment (klutchell) [inline go.mod:22]: What specific go commands did you run to adjust these dependencies, just for my own reference?

**Comment by klutchell** [comment]
@balena-ci retest this please

**Comment by klutchell** [comment]
@balena-ci retest

File: go.mod (hunk 1 above)
Comment (alexgg) [reply go.mod:22]: These are automatically updated. Basically I:

* removed `go.mod`
* Initialized the project again
```
go mod init github.com/balena-os/hostapp
```
* Updated dependencies
```
go mod tidy
```
* Adjusted the docker version
```
go get github.com/docker/docker@<sha1>
```
* Updated dependencies again

**Comment by klutchell** [comment]
@balena-ci retest

**Comment by klutchell** [comment]
I fixed the CI dockerd issue, but now the build itself is failing:
```
 => ERROR [gobuild 5/5] RUN --mount=target=.     --mount=type=cache,target=/go/pkg     --mount=type=cache,target=/root/.cache        make DEST=/out mobynit hostapp.test                                                                                                                                                                                                                                                                                                                                       1.1ss
25h------                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
 > [gobuild 5/5] RUN --mount=target=.     --mount=type=cache,target=/go/pkg     --mount=type=cache,target=/root/.cache        make DEST=/out mobynit hostapp.test:                                                                                                                                                                                                                                                                                                                                                  
#14 1.072 make: /bin/sh: Operation not permitted                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
#14 1.072 make: *** [Makefile:9: mobynit] Error 127
#14 1.072 go build -o /out/mobynit \
#14 1.072 	-ldflags "-extldflags "-static" -s -w" \
#14 1.072 	-tags "netgo osusergo static_build no_btrfs no_cri no_devmapper no_zfs exclude_disk_quota exclude_graphdriver_btrfs exclude_graphdriver_devicemapper exclude_graphdriver_zfs" \
#14 1.072 	./cmd/mobynit
```
https://ci.product-os.io/builds/760716

**Comment by klutchell** [comment]
All the tests passed on this PR, for both x86 and raspberrypi4.
https://github.com/balena-os/meta-balena/pull/2595

Was there anything else we wanted to test manially?

**Comment by alexgg** [comment]
> Was there anything else we wanted to test manially?

Mobynit is used to mount the rootfs on boot, and also on HUP. Ideally some long run testing booting, and a few hundred HUPs would give us the confidence we need.

We could merge this and update meta-balena, and let the automation work on all PRs for some time. We should focus our efforts on bringing as many testbot device types to a working state, and if possible delay releases until we are confident a specific release type is sufficiently tested.

Also, we could always revisit this once we move to a newer LTS Yocto version with a newer libc.

**Comment by alexgg** [comment]
@balena-ci I self-certify!

**Pull request merged**